Automattic/wp-calypso
# LPM API Migration: Begin replacing the Source API version of AliPay with PaymentIntent API version

Related to Automattic/payments-shilling#2976

## Proposed Changes

In D160912-code, we enabled a new version of the Alipay payment method which, unlike the currently available one, uses the Stripe redirect API with a Payment Intent instead of the Stripe Sources API, the latter of which is deprecated and will be disabled soon.

This PR enables the new payment method in calypso checkout, but only for local development (calypso.localhost).

## Testing Instructions

- Sandbox the API so that the purchase will be made in test mode.
- Add a product to your cart and visit checkout using localhost calypso (the calypso.live links won't work for this since that environment is not enabled).
- Inside checkout, set your country code to "China" and any valid postal code (like `510000`).
- In the payment method step, verify that you see "Alipay" as an option.
- Select Alipay, enter any name in the field, and submit the payment.
- When you see the Stripe confirmation page, verify that at the top of the page it reads "Alipay" and mentions "payment_intent" somewhere.
- Authorize the purchase and verify that you are taken back to calypso with a successful purchase (it may take a while).
- Verify on the backend that the order was not made with "stripe-source".
